## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454208`
Commit: [`45baa42484e96a88d92d26c8db8f51c294cd0a87`](https://github.com/NixOS/nixpkgs/commit/45baa42484e96a88d92d26c8db8f51c294cd0a87) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/45baa42484e96a88d92d26c8db8f51c294cd0a87..pull/454208/head))
Merge: [`82cff64c3f68a314489e92aa3fdc3479267b2d3d`](https://github.com/NixOS/nixpkgs/commit/82cff64c3f68a314489e92aa3fdc3479267b2d3d)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/18686248985


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>imagelol</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>imagelol</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>imagelol</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>imagelol</summary>
<pre>/nix/store/4mn3pvf8ahm21qjdh4r2w1lf90gcv9yq-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:210:7: note: to match this &#x27;(&#x27;
/nix/build/nix-8884-1678212189/source/External/zlib/zutil.h:140:33: note: expanded from macro &#x27;fdopen&#x27;
  140 | #        define fdopen(fd,mode) NULL /* No fdopen() */
      |                                 ^
/nix/store/kihcn1ppp1bi8gv7hncwkaalyjx3lzdz-clang-wrapper-21.1.2/resource-root/include/__stddef_null.h:26:14: note: expanded from macro &#x27;NULL&#x27;
   26 | #define NULL ((void*)0)
      |              ^
/nix/build/nix-8884-1678212189/source/External/zlib/zutil.c:133:22: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  133 | const char * ZEXPORT zError(err)
      |                      ^
/nix/build/nix-8884-1678212189/source/External/zlib/zutil.c:305:22: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  305 | voidpf ZLIB_INTERNAL zcalloc (opaque, items, size)
      |                      ^
/nix/build/nix-8884-1678212189/source/External/zlib/zutil.c:315:20: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  315 | void ZLIB_INTERNAL zcfree (opaque, ptr)
      |                    ^
4 warnings and 3 errors generated.
make[2]: *** [External/zlib-no-examples/CMakeFiles/zlib.dir/build.make:275: External/zlib-no-examples/CMakeFiles/zlib.dir/__/zlib/zutil.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:253: External/zlib-no-examples/CMakeFiles/zlib.dir/all] Error 2
make: *** [Makefile:91: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>imagelol</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>imagelol</summary>
<pre>/nix/store/ybygcq3s191f07bwgxbqpgqv0ak4h17k-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:210:7: note: to match this &#x27;(&#x27;
/nix/build/nix-5951-2426206950/source/External/zlib/zutil.h:140:33: note: expanded from macro &#x27;fdopen&#x27;
  140 | #        define fdopen(fd,mode) NULL /* No fdopen() */
      |                                 ^
/nix/store/xbi504848gn164lwzv2x2rclwvjxw7w1-clang-wrapper-21.1.2/resource-root/include/__stddef_null.h:26:14: note: expanded from macro &#x27;NULL&#x27;
   26 | #define NULL ((void*)0)
      |              ^
/nix/build/nix-5951-2426206950/source/External/zlib/zutil.c:133:22: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  133 | const char * ZEXPORT zError(err)
      |                      ^
/nix/build/nix-5951-2426206950/source/External/zlib/zutil.c:305:22: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  305 | voidpf ZLIB_INTERNAL zcalloc (opaque, items, size)
      |                      ^
/nix/build/nix-5951-2426206950/source/External/zlib/zutil.c:315:20: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
  315 | void ZLIB_INTERNAL zcfree (opaque, ptr)
      |                    ^
4 warnings and 3 errors generated.
make[2]: *** [External/zlib-no-examples/CMakeFiles/zlib.dir/build.make:275: External/zlib-no-examples/CMakeFiles/zlib.dir/__/zlib/zutil.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:253: External/zlib-no-examples/CMakeFiles/zlib.dir/all] Error 2
make: *** [Makefile:91: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/im/imagelol/package.nix
+++ b/pkgs/by-name/im/imagelol/package.nix
@@ -37,6 +37,12 @@ stdenv.mkDerivation rec {
     mv imagelol src
     substituteInPlace CMakeLists.txt \
       --replace 'add_subdirectory("imagelol")' 'add_subdirectory("src")'
+
+    substituteInPlace External/zlib-no-examples/CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 2.4.4)" "cmake_minimum_required(VERSION 3.10)"
+    substituteInPlace External/libpng/CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 3.1)" "cmake_minimum_required(VERSION 3.10)" \
+      --replace-fail "cmake_policy(VERSION 3.1)" "cmake_policy(VERSION 3.10)"
   '';
 
   nativeBuildInputs = [ cmake ];
@@ -59,6 +65,7 @@ stdenv.mkDerivation rec {
     license = licenses.mit;
     maintainers = [ ];
     platforms = platforms.unix;
+    broken = stdenv.hostPlatform.isDarwin;
     mainProgram = "ImageLOL";
   };
 }

